src/common/capio/logger.hpp

```diff
@@ -83,11 +82,9 @@ class Logger {
         if (!logfile.is_open()) {
             // NOTE: should never get to this point as capio_server opens up the log file while
             // parsing command line arguments. This is only for failsafe purpose
-            auto hostname = new char[HOST_NAME_MAX];
-            gethostname(hostname, HOST_NAME_MAX);
-            logfile.open(std::string(CAPIO_LOG_SERVER_DEFAULT_FILE_NAME) + hostname + ".log",
+            logfile.open(std::string(CAPIO_LOG_SERVER_DEFAULT_FILE_NAME) + std::to_string(tid) +
```

Comment:
Here you are already passing the `tid` to the filename.
Please move this logic to #33 which is the correct place to discuss it